### PR TITLE
debezium/dbz#2655 Treat blank database.ssl.keystore/truststore as unset in binlog SSL setup

### LINKED ISSUE #2655: MySQL/MariaDB (binlog): blank database.ssl.keystore/truststore is treated as a real path and fails the task
## Bug report

**What Debezium connector do you use and what version?**

MySQL and MariaDB connectors (the shared `debezium-connector-binlog` module). Observed on 3.2.x; the affected code is unchanged on `main`.

---

**What is the connector configuration?**

`database.ssl.mode` set to a value that enables SSL (e.g. `preferred`, `required`, `verify_ca`, `verify_identity`), with `database.ssl.keystore` (or `database.ssl.truststore`) present but set to an empty string `""` instead of being omitted. This happens easily when the value is templated and rendered blank rather than left unset.

---

**What is the captured database version and mode of deployment?**

Not database-version specific. This is purely connector-side config handling in the binlog SSL setup, reproducible regardless of the MySQL/MariaDB version or deployment.

---

**What behavior do you expect?**

A null or blank keystore/truststore value should be treated as "not configured" and ignored, exactly like an omitted value. An empty string should not be opened as a file path.

---

**What behavior do you see?**

The task fails during binlog SSL setup with `DebeziumException("Error loading keystore")` caused by a `FileNotFoundException`, because the empty string is treated as a real path.

Root cause: in `BinlogStreamingChangeEventSource#getBinlogSslSocketFactory` the [keystore guard](https://github.com/debezium/debezium/blob/d6901d740ecd8e357e528388c0f925479b9718b4/debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java#L1573-L1585) is `if (keyFilename != null)` (and the [same for the truststore](https://github.com/debezium/debezium/blob/d6901d740ecd8e357e528388c0f925479b9718b4/debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java#L1589-L1591)). An empty string passes the null check, so [loadKeyStore](https://github.com/debezium/debezium/blob/d6901d740ecd8e357e528388c0f925479b9718b4/debezium-connector-common/src/main/java/io/debezium/jdbc/JdbcConnection.java#L1837-L1846) runs `new FileInputStream("")`, which throws `FileNotFoundException`, caught as `IOException` and rethrown as `DebeziumException`.

Proposed fix: replace the null-only checks with `!Strings.isNullOrBlank(...)` so a blank path is ignored rather than opened. Happy to send a PR with a test.

---

**Do you see the same behaviour using the latest released Debezium version?**

The affected code is unchanged on `main` (the permalinks above are pinned to the current main commit), so the latest code carries the same null-only guard. I reproduced on 3.2.x and have not run a fresh repro on the latest release, but the code path is identical.

---

**Do you have the connector logs, ideally from start till finish?**

Can provide a sanitized log on request. The failure surfaces as `DebeziumException: Error loading keystore` with a `FileNotFoundException` cause during SSL socket factory setup at task start.

---

**How to reproduce the issue using our tutorial deployment?**

Configure a MySQL/MariaDB connector with an SSL-enabling `database.ssl.mode` and set `database.ssl.keystore=""` (an empty string rather than omitting it). The task fails at startup with the error above.

### PR DESCRIPTION
**Pull request by sangeet259**
Fixes debezium/dbz#2655

## Description

When `database.ssl.keystore` or `database.ssl.truststore` is configured as an empty string rather than left unset, `BinlogStreamingChangeEventSource#getBinlogSslSocketFactory` treats the empty value as a real path. The guard checks only for `null`, so a blank string passes it and `loadKeyStore` runs `new FileInputStream("")`, which throws `FileNotFoundException`. The exception is caught as `IOException` and rethrown as `DebeziumException("Error loading keystore")`, failing the task.

This change treats a null or blank value as unset by using `Strings.isNullOrBlank` for both the keystore and truststore paths, so an empty value is ignored instead of being opened. This matches the behavior of an omitted value.

## Testing

SSL is exercised through the MySQL integration tests, and this change only relaxes a `null` check to also treat a blank path as unset, so no new automated test is included. I am happy to add an integration test that starts the connector with an empty keystore path if that is preferred.

Building the connector module on `main` with `./mvnw -B -pl debezium-connector-binlog -am -DskipTests clean install` succeeds, and the formatter and import-order plugins produce no changes beyond the fix.

## PR Checklist
- [x] I have read the [contribution guidelines](https://github.com/debezium/debezium/blob/main/CONTRIBUTING.md) and the [governance document](https://github.com/debezium/governance/blob/main/GOVERNANCE.md) on PR expectations.
- [x] Minimal changes to code not directly related to your change (e.g. no unnecessary formatting changes or refactoring to existing code)
- [x] One feature/change per PR unless tightly coupled
- [x] Do a rebase on upstream `main`

### CODE REVIEW

**Review by vjuranek — COMMENTED** [review]
The change LGTM, would you be able to add a small test for it @sangeet259 ?

**Comment by sangeet259** [comment]
Hi @vjuranek, done. Added a unit test in BinlogStreamingChangeEventSourceTest. Had to widen `getBinlogSslSocketFactory` to `package-private`.

File: debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import io.debezium.connector.binlog.jdbc.BinlogConnectorConnection;
+import io.debezium.connector.binlog.jdbc.ConnectionConfiguration;
+
+/**
+ * Unit tests for {@link BinlogStreamingChangeEventSource#getBinlogSslSocketFactory}.
+ */
+public class BinlogStreamingChangeEventSourceTest {
+
+    @Test
```
Comment (vjuranek) [inline debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSourceTest.java:34]: One more thing, please add `@FixFor("debezium/dbz#2655")` annotation to the tests. Thanks!

File: debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java [hunk 2]
```diff
@@ -1559,7 +1560,8 @@ else if (eventDeserializationFailureHandlingMode == EventProcessingFailureHandli
 
     protected abstract SSLMode sslModeFor(SecureConnectionMode mode);
 
-    private SSLSocketFactory getBinlogSslSocketFactory(BinlogConnectorConfig connectorConfig, BinlogConnectorConnection connection) {
+    // Package-private for testing.
```
Comment (vjuranek) [inline debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java:1563]: Please replace with `@VisibleForTesting` annotation.

File: debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSourceTest.java (hunk 1 above)
Comment (sangeet259) [reply debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSourceTest.java:35]: Added to both tests, thanks for pointing.

File: debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java (hunk 2 above)
Comment (sangeet259) [reply debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java]: changed, thanks.

**Review by vjuranek — APPROVED** [review]
LGTM, let's wait for CI

**Comment by vjuranek** [comment]
Merged, thanks @sangeet259 !
